Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs

```diff
@@ -1314,42 +1370,60 @@ commonTextView state push text' color' fontStyle marginTop =
-      ]
+    [ width MATCH_PARENT
+    , height WRAP_CONTENT
+    , orientation VERTICAL
+    , visibility $ boolToVisibility  state.data.config.showHamMenu
+    , margin $ MarginTop safeMarginTop
+    , accessibility if state.data.settingSideBar.opened /= SettingSideBar.CLOSED || state.props.currentStage == ChatWithDriver || state.props.isCancelRide || state.props.isLocationTracking || state.props.callSupportPopUp || state.props.cancelSearchCallDriver || state.props.showCallPopUp || state.props.emergencyHelpModal || state.props.showRateCard || state.data.waitTimeInfo then DISABLE_DESCENDANT else DISABLE
+    ]
+    $ []
+    <> ( case state.data.followers of
+          Nothing -> []
+          Just followers -> if (showFollowerBar followers state) && (any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithDriver]) then [ followRideBar push followers (MATCH_PARENT) ] else []
+      )
+    <> ( [ linearLayout
+            [ width MATCH_PARENT
+            , height WRAP_CONTENT
+            , orientation HORIZONTAL
+            , visibility if state.data.config.showHamMenu then VISIBLE else GONE
+            , margin $ Margin 16 20 0 0
+            , accessibility if state.data.settingSideBar.opened /= SettingSideBar.CLOSED || state.props.currentStage == ChatWithDriver || state.props.isCancelRide || state.props.isLocationTracking || state.props.callSupportPopUp || state.props.cancelSearchCallDriver || state.props.showCallPopUp || state.props.emergencyHelpModal || state.props.showRateCard || state.data.waitTimeInfo then DISABLE_DESCENDANT else DISABLE
+            ]
+            [ linearLayout
+                [ height $ V 48
+                , width $ V 48
+                , stroke ("1," <> Color.grey900)
+                , background Color.white900
+                , gravity CENTER
+                , cornerRadius 24.0
+                , visibility $ boolToVisibility $ not (any (_ == state.props.currentStage) [ FindingEstimate, ConfirmingRide, FindingQuotes, TryAgain, RideCompleted, RideRating, ReAllocated ])
+                , clickable true
+                , onClick push $ if (any (_ == state.props.currentStage) [ SettingPrice, ConfirmingLocation, PricingTutorial, DistanceOutsideLimits ]) then const BackPressed else const OpenSettings
+                , accessibilityHint if (any (_ == state.props.currentStage) [ SettingPrice, ConfirmingLocation, PricingTutorial, DistanceOutsideLimits ]) then "Back : Button" else "Menu : Button"
+                , accessibility ENABLE
+                , rippleColor Color.rippleShade
+                ]
+                [ imageView
+                    [ imageWithFallback if (any (_ == state.props.currentStage) [ SettingPrice, ConfirmingLocation, PricingTutorial, DistanceOutsideLimits ]) then fetchImage FF_COMMON_ASSET "ny_ic_chevron_left" else if state.data.config.dashboard.enable && (checkVersion "LazyCheck") then fetchImage FF_ASSET "ic_menu_notify" else fetchImage FF_ASSET "ny_ic_hamburger"
```

Comment:
We can move these to variables and use them here. Better to keep the view function clean. Please update this wherever possible (`onClick`, `accessibility`)